Echo.Process/Session/SessionSync.cs

```diff
@@ -108,7 +108,7 @@ public int Incoming(SessionAction incoming)
         /// <param name="sessionId"></param>
         /// <returns></returns>
         internal Option<SessionId> GetSessionId(SupplementarySessionId sessionId) =>
-            suppSessions.Find(sessionId);
+            suppSessions.Where(sid => sid == sessionId).Map(t => t.Key).HeadOrNone();
```

Comment:
This is far too slow if there are 100,000s of sessions.  We need a better solution.

Reply:
Maybe two maps then - SessionId->SuppSess and SuppSess->SessionId. Probably the same goes for cluster implementation: SessionManager.cs line 182 in this PR

Reply:
changed to use two maps both in memory and cluster